exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java

```diff
@@ -207,6 +209,28 @@ private void javaPropertiesToSystemOptions() {
   }
 
   /**
+   * Disallow registering drillbit when:
+   * 1. version is unknown;
+   * 2. drillbit with different version has been already registered.
```

Comment:
Agreed.  In addition to potential soft enforcement, I think we should consider the differences of a version defined in a pom.xml, the version of the running drillbit, and the version of a drill cluster.  By using a setting in drill-override to specify what the drill cluster version is, we don't get accident "first bit to register is the wrong version" situations.  Are there situations where we might want to allow a specific Drill version + all minor versions at that level, or Drill version + 1 Major version for rolling upgrades?